Assets/MixedRealityToolkit.Services/CameraSystem/MixedRealityCameraSystem.cs

```diff
@@ -63,12 +67,39 @@ public MixedRealityCameraProfile CameraProfile
                 return cameraProfile;
             }
         }
+        
+        /// <inheritdoc/>
+        public Camera Main
+        {
+            get
+            {
+                if (main == null || !main.isActiveAndEnabled)
+                {
```

Comment:
This seems like it may break if someone actively changes the main camera without disabling cameras. Compared to checking isActiveAndEnabled, should we check whether the camera is rendering to the main screen? It seems like we may also want to pick the last camera in the render pass because I think you can have multiple cameras so that the first cameras pass is composited into the second.

Reply:
we may also need to check things like [https://docs.unity3d.com/ScriptReference/Camera-targetDisplay.html](https://docs.unity3d.com/ScriptReference/Camera-targetDisplay.html)

It may be worth avoiding obtaining a camera main for android and ios until we have an understanding of what it should return. I'm not sure how ARCore/ARKit transition from the default unity main camera the tracking camera.

---
In reply to: [293876592](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4455#discussion_r293876592) [](ancestors = 293876592)

Reply:
Shelving this discussion for now but let's bring it back up in the camera setup issue we open. Finding the 'right' main camera is a big topic.